plugin/path_impersonated_account_secrets.go

```diff
@@ -45,10 +58,28 @@ func (b *backend) pathImpersonatedAccountAccessToken(ctx context.Context, req *l
 		return nil, err
 	}
 
+	config, err := getConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
 	tokenSource, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
 		TargetPrincipal: acct.EmailOrId,
 		Scopes:          acct.TokenScopes,
+		Lifetime:        config.MaxTTL,
 	}, option.WithCredentials(creds))
+
+	// If the config has been misproperly specified or the organizational policy constraint is
+	// not set up correctly this will allow things to gracefully fallback to a 1 hour token.
+	// The utility function isGoogleApiErrorWithCodes can't be used with the error we get back,
+	// so instead relying on a simple string search in the error code that shouldn't ever be
+	// affected by localization in the error message.
+	if isOrgPolicyConstraintMissingError(err) {
+		tokenSource, err = impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: acct.EmailOrId,
+			Scopes:          acct.TokenScopes,
+		}, option.WithCredentials(creds))
+	}
```

Comment:
This fallback is something we can talk about, but I would really prefer to have the code safely fall back to something that works if there is a misconfiguration somewhere.

Reply:
sounds reasonable :+1:

Reply:
Just to confirm, the misconfiguration that would result in this fallback is only if the organizational policy was not configured with:

```hcl
  list_policy {
    allow {
      all = true
    }
  }
```
(or obviously adding each GCP serviceaccount to the org policy allowlist).

This seems appropriate. Should we at least log a warning when we fallback?

Reply:
I was wondering what we should do if we fall back, good callout.  I can add a log message.